NODEJS-664 Doc updates missed in 4.7.0 release (#414)
datastax/nodejs-driver: README.md

## Compatibility

> The driver supports all versions of Node.js, Cassandra and DSE which are not EOL at the time of release.  For Node.js we support only

Reviewer: ```suggestion
The driver supports all versions of Node.js, Cassandra, and DSE that are not EOL at the time of release. 
Only LTS eligible branches (i.e. even numbered releases) are supported for Node.js.
```

> LTS eligible branches (i.e. even numbered releases).  You can find more information about the Node.js release cycle in the [project documentation].  Consistent with this policy the current version of the driver offers support for the following:

Reviewer: ```suggestion
See the [project documentation] for more information about the Node.js release cycle. 
The current version of the driver offers support consistent with this policy for the following:
```

Author: I might wind up tweaking this one just slightly... let me see what it looks like in context.

## Compatibility

The driver supports all versions of Node.js, Cassandra, and DSE that are not EOL at the time of release.  Only LTS eligible branches (i.e. even numbered releases) are supported for Node.js.  See the [project documentation] for more information about the Node.js release cycle.

The current version of the driver offers support consistent with this policy for the following:

- Apache Cassandra versions 3.0 and above.
- DataStax Enterprise versions 5.1 and 6.8.

> - Node.js versions 18.x and 20.x.

Author: @emeliawilkinson24 This preserves the content changes you recommended... I just moved a few of the sentences together into a single paragraph since all the points seemed to be on related content and it reads a bit more cleanly this way.

Just wanted to make sure you're good with what's here.
